File: rxbinding-leanback-v17/src/main/java/com/jakewharton/rxbinding2/support/v17/leanback/widget/SearchBarSearchQueryEvent.java
package com.jakewharton.rxbinding2.support.v17.leanback.widget;

import android.support.annotation.NonNull;
import android.support.v17.leanback.widget.SearchBar;

public abstract class SearchBarSearchQueryEvent {
  SearchBarSearchQueryEvent() {
  }

  /** The view from which this event occurred. */
  @NonNull public abstract SearchBar view();
  @NonNull public abstract String searchQuery();
}

File: rxbinding-leanback/src/main/java/com/jakewharton/rxbinding4/leanback/SearchBarSearchQueryEvent.kt
package com.jakewharton.rxbinding4.leanback

import androidx.leanback.widget.SearchBar

sealed class SearchBarSearchQueryEvent {
  /** The view from which this event occurred.  */
  abstract val view: SearchBar
  abstract val searchQuery: String
}

data class SearchBarSearchQueryChangedEvent(
  override val view: SearchBar,
  override val searchQuery: String
) : SearchBarSearchQueryEvent()

data class SearchBarSearchQueryKeyboardDismissedEvent(
  override val view: SearchBar,
  override val searchQuery: String
) : SearchBarSearchQueryEvent()

data class SearchBarSearchQuerySubmittedEvent(
  override val view: SearchBar,
  override val searchQuery: String
) : SearchBarSearchQueryEvent()
